File: server/routes/authRoutes.js
const express = require('express');
const router = express.Router();
const { authController } = require('../controllers');

router.post('/signup', authController.signup);
router.post('/login', authController.login);
router.post('/forgot-password', authController.forgotPassword);
router.post('/reset-password', authController.resetPassword);
router.post('/getUserData', authController.authenticateToken, authController.getUserData);

module.exports = router;

File: server/router/authRoutes.go
// auth
package routes

import (
	"Ezhire/controllers"

	"github.com/gorilla/mux"
)

// AuthRoutes sets up the routes for authentication and password management
func AuthRoutes(router *mux.Router) {
	router.HandleFunc("/signup", controllers.Signup).Methods("POST")
	router.HandleFunc("/login", controllers.Login).Methods("POST")
	router.HandleFunc("/forgot-password", controllers.ForgotPassword).Methods("POST")
	router.HandleFunc("/reset-password", controllers.ResetPassword).Methods("POST")
	router.HandleFunc("/user-data", controllers.GetUserData).Methods("GET")
}
